src/components/App/App.jsx

```diff
@@ -111,6 +114,20 @@ class App extends React.Component {
     }
   }
 
+  updateTasksInBackground() {
+    if (!GitHubAuth.isAuthenticated()) return
+
+    let promise = Promise.resolve()
+    this.props.filters.forEach(filter => {
+      promise = promise.then(() => HelperActions.updateTasks(this.props.dispatch, filter))
```

Comment:
Interesting, so this is chaining them all off of each other in `then` callbacks? What I've done before is to use [Promise.all](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all) on an array of promises. So you could do `this.props.filters.map(filter => HelperActions.updateTasks(this.props.dispatch, filter))` if updateTasks returns a promise, and pass the resulting array to Promise.all.